### PR TITLE
CLOUDP-112244: [Milestone] Update Makefile

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-112244

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongocli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongocli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

## Further comments

```bash 
make

setupgolangcilint              Install golangci-lint
deps                           Download go module dependencies
setup                          Set up dev env
link-git-hooks                 Install git hooks
fmt                            Format changed go
lint                           Run linter
fix-lint                       Fix linting errors
check                          Run tests and linters
gen-mocks                      Generate mocks
gen-docs                       Generate docs for commands
build                          Generate a binary for mongocli
build-all                      Generate a binary for both CLIs
build-mongocli                 Generate a mongocli binary in ./bin
build-atlascli                 Generate a atlascli binary in ./bin
build-debug                    Generate binaries in ./bin for debugging both CLIs
build-mongocli-debug           Generate a binary in ./bin for debugging mongocli
build-atlascli-debug           Generate a binary in ./bin for debugging atlascli
integration-test               Run integration tests
unit-test                      Run unit-tests
install                        Install binaries in $GOPATH/bin for both CLIs
install-mongocli               Install mongocli binary in $GOPATH/bin
install-atlascli               Install atlascli binary in $GOPATH/bin
list                           List all make targets

```

### CODE REVIEW

File: build/ci/evergreen.yml [hunk 1]
```diff
@@ -42,6 +42,26 @@ functions:
         env:
           <<: *go_options
         command: make build
```
Comment (andreaangiolillo) [inline build/ci/evergreen.yml:44]: not the evergreen task `"build"` builds both clis

**Review by gssbzn — CHANGES_REQUESTED** [review]
homebrew installs via `make build` makeing this a breaking change, we need to keep this backward-compatible or we need some coordination for our next release

**Comment by andreaangiolillo** [comment]
@gssbzn I will leave the build commands as it and define `build-all` to build both clis. I will remove the build once we have released both clis. Is this a good option for you?

**Comment by gssbzn** [comment]
> I will leave the build command as it is and define build-all to build both clis. I will update the build once we have released both clis. Is this a good option for you?

Sure that works for me

**Review by blva — APPROVED** [review]
lgtm

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**